assets/src/admin/paired-browsing/app.js

```diff
@@ -0,0 +1,304 @@
+/**
+ * WordPress dependencies
+ */
+import { addQueryArgs, hasQueryArg, removeQueryArgs } from '@wordpress/url';
+/**
+ * Internal dependencies
+ */
+import './app.css';
+
+const { app, history } = window;
+const { ampSlug, ampPairedBrowsingQueryVar, ampValidationErrorsQueryVar } = app;
+
+class PairedBrowsingApp {
+	exitLink;
+
+	/**
+	 * Constructor.
+	 */
+	constructor() {
+		this.nonAmpIframe = document.querySelector( '#non-amp iframe' );
+		this.ampIframe = document.querySelector( '#amp iframe' );
+		this.ampPageHasErrors = false;
+
+		// Link to exit paired browsing.
+		this.exitLink = document.getElementById( 'exit-link' );
+
+		// Overlay that is displayed on the client that becomes disconnected.
+		this.disconnectOverlay = document.querySelector( '.disconnect-overlay' );
+		this.disconnectText = {
+			general: document.querySelector( '.disconnect-overlay .dialog-text span.general' ),
+			invalidAmp: document.querySelector( '.disconnect-overlay .dialog-text span.invalid-amp' ),
+		};
+		this.disconnectButtons = {
+			continue: document.querySelector( '.disconnect-overlay .button.continue' ),
+			exit: document.querySelector( '.disconnect-overlay .button.exit' ),
+			goBack: document.querySelector( '.disconnect-overlay .button.go-back' ),
+		};
+		this.addDisconnectButtonListeners();
+
+		// Load clients.
+		Promise.all( this.getIframeLoadedPromises() );
+	}
+
+	/**
+	 * Add event listeners for buttons on disconnect overlay.
+	 */
+	addDisconnectButtonListeners() {
+		// The 'Continue' button simply hides the 'disconnected' overlay.
+		this.disconnectButtons.continue.addEventListener( 'click', () => {
+			this.disconnectOverlay.classList.remove( 'disconnected' );
+		} );
```

Comment:
I think perhaps the Continue button should result in actually exiting paired browsing. Thoughts?